packages/create-turbo/src/commands/create/index.ts

```diff
@@ -0,0 +1,181 @@
+import { isOnline } from "../../utils/isOnline";
+import { turboGradient, turboLoader, info, error } from "../../logger";
+
+function handleWorkspaceErrors(err: unknown) {
+  if (err instanceof ConvertError && err.type !== "unknown") {
+    error(chalk.red(err.message));
+    process.exit(1);
+  } else {
+    // if it's an unknown error type, re-throw to root
+    throw err;
+  }
+}
+
+const SCRIPTS_TO_DISPLAY: Record<string, string> = {
+  build: "Build",
+  dev: "Develop",
+  test: "Test",
+  lint: "Lint",
+};
+
+export async function create(
+  directory: CreateCommandArgument,
+  packageManager: CreateCommandArgument,
+  opts: CreateCommandOptions
+) {
+  const { skipInstall } = opts;
+  console.log(chalk.bold(turboGradient(`\n>>> TURBOREPO\n`)));
+  info(`Welcome to Turborepo! Let's get you set up with a new codebase.`);
+  console.log();
+
+  const online = await isOnline();
+  if (!online) {
+    error(
+      "You appear to be offline. Please check your network connection and try again."
+    );
+    process.exit(1);
+  }
+  const { root, projectName } = await prompts.directory({ directory });
+  const relativeProjectDir = path.relative(process.cwd(), root);
+  const projectDirIsCurrentDir = relativeProjectDir === "";
+
+  const selectedPackageManagerDetails = await prompts.packageManager({
+    packageManager,
+  });
+
+  const { example, examplePath } = opts;
+
+  const { hasPackageJson, availableScripts } = await createProject({
+    appPath: root,
+    projectName,
+    example: example && example !== "default" ? example : "basic",
+    examplePath,
+  });
+
+  let project: Project = {} as Project;
+  try {
+    project = await getWorkspaceDetails({ root });
+  } catch (err) {
+    handleWorkspaceErrors(err);
+  }
+
+  if (project.packageManager !== selectedPackageManagerDetails.name) {
+    try {
+      await convert({
+        root,
+        to: selectedPackageManagerDetails.name,
+        options: {
+          // skip install after conversion- we will do it later
+          skipInstall: true,
+        },
+      });
+    } catch (err) {
+      handleWorkspaceErrors(err);
+    }
+  }
+
+  info("Created a new Turborepo with the following:");
+  console.log();
+  if (project.workspaceData.workspaces.length > 0) {
+    const workspacesForDisplay = project.workspaceData.workspaces
+      .map((w) => ({
+        group: path.relative(root, w.paths.root).split(path.sep)?.[0] || "",
+        title: path.relative(root, w.paths.root),
+        description: w.description,
+      }))
+      .sort((a, b) => a.title.localeCompare(b.title));
+
+    let lastGroup: string | undefined;
+    workspacesForDisplay.forEach(({ group, title, description }, idx) => {
+      if (idx === 0 || group !== lastGroup) {
+        console.log(chalk.cyan(group));
+      }
+      console.log(
+        ` - ${chalk.bold(title)}${description ? `: ${description}` : ""}`
+      );
+      lastGroup = group;
+    });
+  } else {
+    console.log(chalk.cyan("apps"));
```

Comment:
I would have thought that in the absence of workspaces we'd want to treat the package as a pseudo-workspace with one item in it. I think itd clean out code up a little.

Reply:
Yea I was doing that, but since we do support non-monorepos as first party users of turbo I didn't want to make them also look like monorepos